index.js

```diff
@@ -102,3 +109,43 @@ axios
     setFailed(error.message);
     endGroup();
   });
+
+const pollInterval = 3000; // in milliseconds
+
+let followWorkflow = getInput("Follow").toLowerCase() == "true";
+
+const pollWorkflow = () => {
+  axios
+    .get(workFlowUrl, {
+      headers: headers,
+    })
+    .then((response) => {
+      if (response.data.items[0].status != "running") {
+        followWorkflow = false;
+        if (response.data.items[0].status == "success") {
+          info("CircleCI Workflow is complete");
+        } else {
+          setFailed(
+            `Failure: CircleCI Workflow ${response.data.items[0].status}`
+          );
+        }
+      }
+    })
+    .catch((error) => {
+      startGroup("Failed to Poll CircleCI Workflow");
+      coreError(error);
+      setFailed(error.message);
+      endGroup();
+    });
+};
+
+if (followWorkflow) {
```

Comment:
I think you only need to wrap the `info` call. The `setInterval` should immedately be unset if `followWorkflow` is false.

```typescript
if (followWorkflow) {
  info("Polling CircleCI Workflow");
}

const checkWebsiteStatus = setInterval(() => {
  if (!followWorkflow) {
    clearInterval(checkWebsiteStatus);
  } else {
    pollWorkflow();
  }
}, pollInterval);
```